netty/src/test/java/io/grpc/netty/NettyServerHandlerTest.java

```diff
@@ -147,16 +160,28 @@ public void transportTerminated() {
     }
   }
 
+  /**
+   * Will not run {@code setUp()} automatically if the test name contains "customSetUp".
+   */
   @Before
-  public void setUp() throws Exception {
+  public void maybeRunSetup() throws Exception {
+    if (!testName.getMethodName().contains("customSetUp")) {
```

Comment:
I don't feel great about this, because its magical and can easily make things harder to debug ("why does that test method work and not this one?"). Since it seems so many tests already call `setUp()`, let's remove the `@Before` and have each test call `setUp()` explicitly.